term/src/terminalstate/mod.rs

```diff
@@ -62,6 +62,15 @@ pub(crate) enum MouseEncoding {
     SgrPixels,
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum MouseCursor {
+    Arrow,
+    Hand,
+    Text,
+    SizeUpDown,
+    SizeLeftRight,
+}
```

Comment:
I moved this struct because I figured the terminal state getter was a good place to convert the string from the control sequence to a type understood by the current codebase (see `get_mouse_cursor_shape` in this file).

Not sure if this is the right place though, especially considering that means it needs to be referenced much higher up the stack, and I needed to add a new `wezterm-term` reference to the `window/Cargo.toml` file.